### ISSUE TITLE
`derivable_impls` does not respect MSRV configuration

### ISSUE THREAD

**Issue by balthild** [rust-lang/rust-clippy#16126]
### Summary

<img width="1187" height="979" alt="Image" src="https://github.com/user-attachments/assets/61293583-e78b-4412-b3ec-50b3a7e33fe7" />

### Lint Name

derivable_impls

### Reproducer

I tried this code:

```toml
# Cargo.toml
[package]
# ...
rust-version = "1.85"
```

```toml
# clippy.toml
msrv = "1.85"
```

```rust
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum MessageLevel {
    Info,
    Warning,
    Error,
}

impl Default for MessageLevel {
    fn default() -> Self {
        Self::Info
    }
}
```

I saw this happen:

```
warning: this `impl` can be derived
  --> src/dialog/message.rs:12:1
   |
12 | / impl Default for MessageLevel {
13 | |     fn default() -> Self {
14 | |         Self::Info
15 | |     }
16 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#derivable_impls
   = note: `#[warn(clippy::derivable_impls)]` on by default
help: replace the manual implementation with a derive attribute and mark the default variant
   |
 6 + #[derive(Default)]
 7 ~ pub enum MessageLevel {
 8 ~     #[default]
 9 ~     Info,
   |
```

I expected to see this happen:

No warning should be produced since deriving `Default` is not available in Rust 1.85.

### Version

```text
rustc 1.91.1 (ed61e7d7e 2025-11-07)
binary: rustc
commit-hash: ed61e7d7e242494fb7057f2657300d9e77bb4fcb
commit-date: 2025-11-07
host: aarch64-apple-darwin
release: 1.91.1
LLVM version: 21.1.2
```

**Comment by profetia** [comment]
`Default` derive on `enum` is introduced in 1.62, not 1.85

**Comment by balthild** [comment]
Sorry, I misinterpreted the diagnostic message (thought it were added in 1.91.0).
